App/Device/app_usbd_cdc_ecm.c

```diff
@@ -0,0 +1,430 @@
+*/
+
+
+/*
+*********************************************************************************************************
+*********************************************************************************************************
+*                                           GLOBAL FUNCTION
+*********************************************************************************************************
+*********************************************************************************************************
+*/
+
+/*
+*********************************************************************************************************
+*                                       App_USBD_CDC_ECM_Init()
+*
+* Description : Initialize the USB CDC ECM application.
+*
+* Argument(s) : dev_nbr              Device number.
+*
+*               cfg_hs               High-speed configuration number, set to USBD_CFG_NBR_NONE if not used.
+*
+*               cfg_fs               Full-speed configuration number, set to USBD_CFG_NBR_NONE if not used.
+*
+* Return(s)   : USBH_ERR_NONE,       if application initialized successfully.
+*
+*               Specific error code, otherwise.
+*
+* Note(s)     : None.
+*********************************************************************************************************
+*/
+
+USBD_ERR  App_USBD_CDC_ECM_Init (CPU_INT08U  dev_nbr,
+                                 CPU_INT08U  cfg_hs,
+                                 CPU_INT08U  cfg_fs)
+{
+    USBD_ERR   err;
+    OS_ERR     os_err;
+
+
+    USBD_ECM_Init(&err);
+    if (err != USBD_ERR_NONE) {
+        return (err);
+    }
+
+    App_USBD_ECM_SubclassNbr = USBD_ECM_Add(App_USBD_ECM_MgmtReq, NULL, DEF_TRUE, &err);
+    if (err != USBD_ERR_NONE) {
+        return (err);
+    }
+
+    if (cfg_hs != USBD_CFG_NBR_NONE) {
+        USBD_ECM_CfgAdd(App_USBD_ECM_SubclassNbr,
+                        dev_nbr,
+                        cfg_hs,
+                        App_USBD_ECM_MACAddrStr,
+                        App_USBD_ECM_EthStats,
+                        App_USBD_ECM_MaxSegSize,
+                        App_USBD_ECM_NumMCFilters,
+                        App_USBD_ECM_NumPwrFilters,
+                        &err);
+        if (err != USBD_ERR_NONE) {
+            return (err);
+        }
+    }
+
+    if (cfg_fs != USBD_CFG_NBR_NONE) {
+        USBD_ECM_CfgAdd(App_USBD_ECM_SubclassNbr,
+                        dev_nbr,
+                        cfg_fs,
+                        App_USBD_ECM_MACAddrStr,
+                        App_USBD_ECM_EthStats,
+                        App_USBD_ECM_MaxSegSize,
+                        App_USBD_ECM_NumMCFilters,
+                        App_USBD_ECM_NumPwrFilters,
+                        &err);
+        if (err != USBD_ERR_NONE) {
+            return (err);
+        }
+    }
+
+                                                                /* ----------- TASK FOR SENDING ARP MSG --------------- */
+    OSTaskCreate(            &App_USBD_ECM_TaskTCB,
```

Comment:
Hi @wes-fflores. I notice there's no equivalent to `USBH_OS_TaskCreate` for OS compatibility. Is there a good way to handle OS II and III compatibility here? Or is it okay to make it work only with uCOS-III?